config/routes.rb

```diff
@@ -41,9 +41,9 @@
     post 'users/authenticate-sensitive' => 'users#authenticate_user_for_sensitive_edit'
     delete 'users/sign-out-other' => 'sessions#destroy_other', as: :destroy_other_user_sessions
   end
-  post 'registration/:id/refund/:payment_id' => 'registrations#refund_payment', as: :registration_payment_refund
   post 'registration/:id/load-payment-intent' => 'registrations#load_payment_intent', as: :registration_payment_intent
-  get 'registration/:id/payment-completion' => 'registrations#payment_completion', as: :registration_payment_completion
+  post 'competitions/:competition_id/:payment_id/refund' => 'registrations#refund_payment', as: :registration_payment_refund
```

Comment:
Might I suggest `competitions/:competition_id/refund/:payment_id`?
That way it's clearer which ID is which.